packages/editor/src/store/actions.js

```diff
@@ -788,6 +788,26 @@ export function unlockPostSaving( lockName ) {
 	};
 }
 
+/**
+ * Sets the rich text selection to a certain value.
+ *
+ * @param {string} blockClientId The block that text is selected in.
+ * @param {string} identifier    The RichText identifier that text is selected in.
+ * @param {Object} range         The start and end that text is selected in.
+ *
+ * @return {Object} Action object.
+ */
+export function setRichTextSelection( blockClientId, identifier, range ) {
```

Comment:
While I still think this is valuable, I'm not sure if we're doing it right yet (duplicating block, and onBlur handling). I don't think it's critical for v1 of annotations, can we leave it for its own PR?

Reply:
I've spun this out into it's own PR: https://github.com/WordPress/gutenberg/pull/11658. I will clean that PR up when this lands.